python/tests/kernel/test_kernel_features.py

```diff
@@ -988,3 +990,228 @@ def kernel(argument: float):
+
+    @cudaq.kernel
+    def test_kernel(nQubits: int):
+        qubits = cudaq.qvector(nQubits)
+        crx(2.2, 0, 1)
+
+    with pytest.raises(RuntimeError) as e:
+        test_kernel.compile()
+    assert 'control operand 0 is not of quantum type' in repr(e)
+
+    @cudaq.kernel
+    def test_kernel(nQubits: int):
+        qubits = cudaq.qvector(nQubits)
+        mz(22)
+
+    with pytest.raises(RuntimeError) as e:
+        test_kernel.compile()
+    assert 'target operand 0 is not of quantum type' in repr(e)
+
+    @cudaq.kernel
+    def test_kernel(nQubits: int):
+        qubits = cudaq.qvector(nQubits)
+        rx.ctrl(2.2, 2, 3)
+
+    with pytest.raises(RuntimeError) as e:
+        test_kernel.compile()
+    assert 'control operand 0 is not of quantum type' in repr(e)
+
+    @cudaq.kernel
+    def test_kernel(nQubits: int):
+        qubits = cudaq.qvector(nQubits)
+        sdg(2)
+
+    with pytest.raises(RuntimeError) as e:
+        test_kernel.compile()
+    assert 'target operand 0 is not of quantum type' in repr(e)
+
+    @cudaq.kernel
+    def test_kernel(nQubits: int):
+        qubits = cudaq.qvector(nQubits)
+        x.adj(3)
+
+    with pytest.raises(RuntimeError) as e:
+        test_kernel.compile()
+    assert 'target operand 0 is not of quantum type' in repr(e)
+
+    @cudaq.kernel
+    def test_kernel(nQubits: int):
+        qubits = cudaq.qvector(nQubits)
+        swap.ctrl(3, 4)
+
+    with pytest.raises(RuntimeError) as e:
+        test_kernel.compile()
+    assert 'target operand 0 is not of quantum type' in repr(e)
+
+    @cudaq.kernel
+    def test_kernel(nQubits: int):
+        qubits = cudaq.qvector(nQubits)
+        rx.ctrl(1.1, 3, 2)
+
+    with pytest.raises(RuntimeError) as e:
+        test_kernel.compile()
+    assert 'control operand 0 is not of quantum type' in repr(e)
+
+    @cudaq.kernel
+    def test_kernel(nQubits: int):
+        qubits = cudaq.qvector(nQubits)
+        rx.adj(2.2, 3)
+
+    with pytest.raises(RuntimeError) as e:
+        test_kernel.compile()
+    assert 'target operand 0 is not of quantum type' in repr(e)
+
+
+def test_math_module_pi_1448():
+    import math
+    @cudaq.kernel
+    def test_kernel() -> float:
+        theta = math.pi
+        return theta
+    test_kernel.compile()
+    assert np.isclose(test_kernel(), math.pi, 1e-12)
+
+def test_len_qvector_1449():
+
+    @cudaq.kernel
+    def test_kernel(nCountingQubits: int) -> int:
+        qubits = cudaq.qvector(nCountingQubits)
+        # N = counting_qubits.size()
+        N = len(qubits)
+        h(qubits)
+        return N
+
+    test_kernel.compile()
+    assert test_kernel(5) == 5
+
+def test_missing_paren_1450():
+
+    @cudaq.kernel
+    def test_kernel():
+        state_reg = cudaq.qubit
+        x(state_reg)
+    
+    with pytest.raises(RuntimeError) as e:
+        test_kernel.compile()
+    assert 'Invalid assignment detected.'
+
+# def test_cast_error_1451():
```

Comment:
Why is this new test for issue #1451 commented out?

Reply:
Whoops, oversight. Adding back in next commit.